### PR TITLE
filters: Add test/server:filter_config_test

### PR DESCRIPTION
**Pull request by auni53**
Commit Message: Add unit test for NamedHttpFilterConfigFactory
Additional Description: As part of #14470, I'll be modifying the base filter interface to include an overrideable `dependencies()` function. This is prep work. 
Risk Level: Low (test only)

### CODE REVIEW

**Comment by asraa** [comment]
Could you please check format? (Run `tools/code_format/check_format.py fix` locally)

**Comment by auni53** [comment]
Ah oops, thought I'd set that up to run automatically.

File: test/server/filter_config_test.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "test/mocks/server/factory_context.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class TestHttpFilter : public Http::StreamDecoderFilter {
```
Comment (zuercher) [inline test/server/filter_config_test.cc:11]: Maybe extend the PassThroughDecoderFilter in extensions/filters/http/common? One less place to update if filters get a new virtual method.

**Review by zuercher — APPROVED** [review]

File: test/server/filter_config_test.cc (hunk 1 above)
Comment (auni53) [reply test/server/filter_config_test.cc]: No need to extend, I just plugged the filter in. I was debating whether we'd anticipate future changes needed, but actually most of the future work here will happen at the FilterFactory level so left that stuff in place.

**Review by zuercher — APPROVED** [review]

**Pull request merged**